### PR TITLE
Movies Refactor

### PR DESCRIPTION
**Pull request by angy113**
@carledriss ready for code review

### CODE REVIEW

File: Movies/main/java/org/fundacionjala/movies/Main.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.fundacionjala.movies;
+
+public class Main {
```
Comment (carledriss) [inline Movies/main/java/org/fundacionjala/movies/Main.java]: Remove this unnecessary class

File: Movies/main/java/org/fundacionjala/movies/Movie.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package org.fundacionjala.movies;
+
+public abstract class Movie {
```
Comment (carledriss) [inline Movies/main/java/org/fundacionjala/movies/Movie.java]: Review your implementation again because you can still refactor code.

**Comment by angy113** [comment]
@carledriss ready to code review

File: Movies/main/java/org/fundacionjala/movies/Movie.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    private final String _title;
```
Comment (carledriss) [inline Movies/main/java/org/fundacionjala/movies/Movie.java]: Why are you using _ in the attributes?

File: Movies/main/java/org/fundacionjala/movies/Movie.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    private final double _amount;
+    private final double _increment;
+    private final int _limitDays;
+
+    private static final int POINT = 1;
```
Comment (carledriss) [inline Movies/main/java/org/fundacionjala/movies/Movie.java:4]: attributes static final are defined first.

File: Movies/main/java/org/fundacionjala/movies/Statement.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Statement {
+
+    private static final String TAB = "\t";
+    private static final String BREAK_LINE = "\n";
+    private final Customer customer;
+    private final List<Rental> rentals = new ArrayList<Rental>();
```
Comment (carledriss) [inline Movies/main/java/org/fundacionjala/movies/Statement.java]: It is not a good practice to do that.
Move the new instance into constructor.

File: Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * Test for {@link ChildrenMovie}
+ */
+public class ChildrenMovieTest {
+
+    private Movie childrenMovie;
+    private static final double PRECISION = 0.0001;
+
+    @Before
+    public void setUp(){
+        String movieName = "Tangled";
```
Comment (carledriss) [inline Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java]: this should be final

File: Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        childrenMovie = new ChildrenMovie(movieName);
+    }
+
+    @Test
+    public void test_theAmountIsCalculatedInChildrenMovie_whenRenterDaysAreMoreThanThree(){
+        double expectedAmount = 3;
```
Comment (carledriss) [inline Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java]: same as above

File: Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        final int renterDays = 4;
+        assertEquals(expectedAmount, childrenMovie.calculateAmount(renterDays), PRECISION);
+    }
+
+    @Test
+    public void test_theAmountIsCalculatedInChildrenMovie_whenRenterDaysAreLessThanThree(){
+        double expectedAmount = 1.5;
```
Comment (carledriss) [inline Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java]: same as above

File: Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        final int renterDays = 2;
+        assertEquals(expectedAmount, childrenMovie.calculateAmount(renterDays), PRECISION);
+    }
+
+    @Test
+    public void test_theFrequentRenterPointsIsCalculatedInChildrenMovie() {
```
Comment (carledriss) [inline Movies/test/java/org/fundacionjala/movies/ChildrenMovieTest.java]: the name should be 
test_methodName_Scenario
in this case
test_calculateFrequentRenterPoints

Review all your tests

File: Movies/test/java/org/fundacionjala/movies/CustomerTest.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import static org.junit.Assert.assertEquals;
+/**
+ * Test for {@link Customer}
+ */
+public class CustomerTest {
+
+    @Test
+    public void test_theCustomerAddedIsTheSame(){
+        Customer customer = new Customer("Angy");
+        assertEquals(customer.getName(),"Angy");
```
Comment (carledriss) [inline Movies/test/java/org/fundacionjala/movies/CustomerTest.java]: It is not necessary to add unit tests for getter and setters.

**Comment by angy113** [comment]
@carledriss ready to code review

**Comment by angy113** [comment]
@carledriss ready to code review

**Comment by carledriss** [comment]
Looks good to me!

**Pull request merged**